docs/content/Details.md

```diff
@@ -66,6 +29,39 @@ Details components get `COMMON` system props. Read our [System Props](/system-pr
 | :- | :- | :-: | :- |
 | defaultOpen | Boolean | | Sets the initial open/closed state |
 | overlay | Boolean | false | Sets whether or not element will close when user clicks outside of it |
-| open | Boolean | | Use the open prop if you'd like to manage the open state |
-| onToggle | Function | | Called whenever user clicks on `summary` element. If you are controlling your own `open` state this will be the only function called on click, otherwise it's called before the internal `handleToggle` function.|
-| onClickOutside | Function | | Function to call whenever user clicks outside of the Details component. This is optional and only necessary if you are controlling your own `open` state. |
+| onToggle | Function | | Called whenever user clicks on `summary` element |
+| ref | React ref | | ref to pass down to Details component |
+
+
+## Details.Context
+Details.Context is a [context object](https://reactjs.org/docs/context.html#reactcreatecontext) that exposes some helpful state values to be used via [`React.useContext`](https://reactjs.org/docs/hooks-reference.html#usecontext) in consuming applications.  Details/Context can only be used in components that are already wrapped in a `Details` as `Details` contains the [context provider](https://reactjs.org/docs/context.html#contextprovider).
+
+`Details.Context` contains the `open` and `setOpen` values which can be used to conditionally update UI based on the open state of the dropdown or set up useEffects to trigger some actions when the `open` state changes.
+
+### Example Usage
+```jsx
+import {Details, Button} from `@primer/components`
+import React, {useContext} from 'react'
+
+const MyDetails = () => {
+  <Details>
+    <MyButton as="summary" />
+    content
+  </Details>
+}
+
+// note that we can only use the context in components that are already wrapped by Details (and thus the Context.Provider)
+const MyButton = () => {
+  const detailsContext = useContext(Details.Context);
+
+  return (
+    <Button as="summary">{detailsContext.open ? 'Open' : 'Closed'}</Button>
```

Comment:
Is it important that this button renders a "summary" tag? Because I can see people easily forgetting to do that, especially now that the component is further from the Details wrapper.

Reply:
Yeah it is required - I have the same concerns! I'm not sure how to put in safeguards around what tag is used without sacrificing flexibility 🤔